framework/src/systems/LinearSystem.C

```diff
@@ -0,0 +1,293 @@
+    ElemInfoRange elem_info_range(_fe_problem.mesh().ownedElemInfoBegin(),
+                                  _fe_problem.mesh().ownedElemInfoEnd());
+
+    using FaceInfoRange = StoredRange<MooseMesh::const_face_info_iterator, const FaceInfo *>;
+    FaceInfoRange face_info_range(_fe_problem.mesh().ownedFaceInfoBegin(),
+                                  _fe_problem.mesh().ownedFaceInfoEnd());
+
+    ComputeLinearFVElementalThread elem_thread(_fe_problem,
+                                               _fe_problem.linearSysNum(name()),
+                                               Moose::FV::LinearFVComputationMode::FullSystem,
+                                               vector_tags);
+    Threads::parallel_reduce(elem_info_range, elem_thread);
+
+    ComputeLinearFVFaceThread face_thread(_fe_problem,
+                                          _fe_problem.linearSysNum(name()),
+                                          Moose::FV::LinearFVComputationMode::FullSystem,
+                                          vector_tags);
+    Threads::parallel_reduce(face_info_range, face_thread);
+  }
+  PARALLEL_CATCH;
+
+  closeTaggedMatrices(matrix_tags);
+
+  _linear_implicit_system.matrix->close();
+  _linear_implicit_system.rhs->close();
+
+  // Accumulate the occurrence of solution invalid warnings for the current iteration cumulative
+  // counters
+  _app.solutionInvalidity().solutionInvalidAccumulation();
+}
+
+NumericVector<Number> &
+LinearSystem::getRightHandSideTimeVector()
+{
+  return *_rhs_time;
+}
+
+NumericVector<Number> &
+LinearSystem::getRightHandSideNonTimeVector()
+{
+  return *_rhs_non_time;
+}
+
+void
+LinearSystem::augmentSparsity(SparsityPattern::Graph & /*sparsity*/,
+                              std::vector<dof_id_type> & /*n_nz*/,
+                              std::vector<dof_id_type> & /*n_oz*/)
+{
+  mooseError("LinearSystem does not support AugmentSparsity!");
+}
+
+void
+LinearSystem::solve()
+{
+  TIME_SECTION("LinearSystem::solve", 2, "Solving linear system");
+
+  // Clear the iteration counters
+  _current_l_its = 0;
+
+  system().solve();
+
+  _n_linear_iters = _linear_implicit_system.n_linear_iterations();
+
+  // store info about the solve
+  _final_linear_residual = _linear_implicit_system.final_linear_residual();
+
+  // determine whether solution invalid occurs in the converged solution
+  _solution_is_invalid = _app.solutionInvalidity().solutionInvalid();
+
+  // output the solution invalid summary
+  if (_solution_is_invalid)
+  {
+    // sync all solution invalid counts to rank 0 process
+    _app.solutionInvalidity().sync();
+
+    if (_fe_problem.allowInvalidSolution())
+      mooseWarning("The Solution Invalidity warnings are detected but silenced! "
+                   "Use Problem/allow_invalid_solution=false to activate ");
+    else
+      // output the occurrence of solution invalid in a summary table
+      _app.solutionInvalidity().print(_console);
+  }
```

Comment:
this stuff could use its own routine since it s shared between systems

Reply:
Okay, I put this in the solver system!